File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        dici={}
        for key in nums:
            if key in dici:
                dici[key]+=1
            else:
                dici[key]=1
        ans=0
        for i in dici:
            n=dici[i]
            temp=n*(n-1)//2
            ans=ans+temp
        return ans

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer, Integer> pair = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            pair.put(nums[i], pair.getOrDefault(nums[i], 0) + 1);
        }
        int ans = 0;
        for (int j : pair.keySet()) {
            int val = pair.get(j);
            int temp = (val * (val - 1) / 2);
            ans = ans + temp;
        }
        return ans;
        
    }
}